sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuColumnarBatchSerializer.scala

```diff
@@ -491,6 +494,115 @@ private class KudoSerializerInstance(
     throw new UnsupportedOperationException
 }
 
+
+
+
+private class KudoGpuSerializerInstance(
+    val metrics: Map[String, GpuMetric],
+    val dataTypes: Array[DataType],
+) extends SerializerInstance {
+  private val serTime = metrics(METRIC_SHUFFLE_SER_STREAM_TIME)
+  private val deserTime = metrics(METRIC_SHUFFLE_DESER_STREAM_TIME)
+
+  override def serializeStream(out: OutputStream): SerializationStream = new SerializationStream {
+
+    override def writeValue[T: ClassTag](value: T): SerializationStream = serTime.ns {
+      val batch = value.asInstanceOf[ColumnarBatch]
+      val numColumns = batch.numCols()
+      withResource(new ArrayBuffer[AutoCloseable](numColumns)) { toClose =>
+        if (batch.numCols() > 0) {
+          val firstCol = batch.column(0)
+          // TODO: handle secondCol
+          firstCol match {
+            case vector: SlicedGpuCompressedColumnVector =>
+              withResource(vector.getWrap.getTableBuffer) { buf =>
+                val start = vector.getStart
+                val len = vector.getEnd - start
+                val data = buf.slice(start, len)
+                val hostBuffer = HostMemoryBuffer.allocate(data.getLength)
+                hostBuffer.copyFromDeviceBuffer(data);
+                val ret: Array[Byte] = new Array(hostBuffer.getLength.toInt)
```

Comment:
This may be very large and I think it is cleaner to do something like 

```
var remaining =  hostBuffer.getLength.toInt
val temp = new Array[Byte](math.min(8192, remaining))
val at = 0
while (remaining > 0) {
  val read = hostBuffer.getBytes(temp, at, math.min(remaining, temp.length))
  out.write(temp, 0, read)
  at = at + read
  remaining = remaining - read
}
```